src/Microsoft.ApiDesignGuidelines.Analyzers/Core/ImplementStandardExceptionConstructors.cs

```diff
@@ -27,23 +31,96 @@ public abstract class ImplementStandardExceptionConstructorsAnalyzer : Diagnosti
                                                                              DiagnosticSeverity.Warning,
                                                                              isEnabledByDefault: false,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
-                                                                             customTags: WellKnownDiagnosticTags.Telemetry);
-        internal static DiagnosticDescriptor AccessibilityRule = new DiagnosticDescriptor(RuleId,
-                                                                             s_localizableTitle,
-                                                                             s_localizableMessageAccessibility,
-                                                                             DiagnosticCategory.Design,
-                                                                             DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
-                                                                             description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/ms182151.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
-        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MissingConstructorRule, AccessibilityRule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(MissingConstructorRule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
-            
+            // Analyze named types 
+            analysisContext.RegisterSymbolAction(symbolContext =>
+            {
+                AnalyzeSymbol(symbolContext.Symbol, symbolContext);
+
+            }, SymbolKind.NamedType);
+        }
+
+        //abstract methods, which the language specific analyzers implements - these will return the required constructor method signatures for CSharp/Basic
+        protected abstract string GetParameterLessConstructorMessage(ISymbol symbol);
```

Comment:
Name these consistently - maybe all of them should end with Signature instead of Message?